From fbd0d6782d02fd2062821cb1dd34b19d7e19c6a2 Mon Sep 17 00:00:00 2001
From: Manjuraehmad Momin <manjuraehmad.momin@amd.com>
Date: Tue, 13 Feb 2024 09:50:56 +0530
Subject: [PATCH 5/5] fix: Runtime PM usage count underflow!

For some reason ox03f10_s_frame_interval()(camera sensor driver)
does a pm_runtime_put(). But it has never done a pm_runtime_get()
leading to the following error being logged into dmesg:
OX03F10 9-0066: Runtime PM usage count underflow!
---
 drivers/media/i2c/ox03f10.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/ox03f10.c b/drivers/media/i2c/ox03f10.c
index 057bc55..512834c 100644
--- a/drivers/media/i2c/ox03f10.c
+++ b/drivers/media/i2c/ox03f10.c
@@ -719,7 +719,9 @@ static int ox03f10_s_frame_interval(struct v4l2_subdev *sd,
 
 unlock:
 	mutex_unlock(&ox03f10->lock);
-	pm_runtime_put(&ox03f10->client->dev);
+	/* commented below code to fix error reported in dmesg:
+	   OX03F10 9-0066: Runtime PM usage count underflow! */
+	//pm_runtime_put(&ox03f10->client->dev);
 
 	return ret;
 }
-- 
1.8.3.1

